uint32 milliseconds
int32 rate
int32[] data

